### PR TITLE
Add channel.content_owner_details

### PR DESCRIPTION
**Pull request by stavro**
The easiest way to confirm that a channel is in a particular network is by checking the content_owner_details of that channel.  The content owner details object is **only accessible** if you are the content owner.

https://developers.google.com/youtube/v3/docs/channels#contentOwnerDetails

``` ruby
content_owner = Yt::ContentOwner.new(access_token: "...", owner_name: "FullScreen")

# channel in FullScreen
channel = Yt::Channel.new(id: 'UC-Z3Pey219Tyr9aW1mX7HQQ', auth: content_owner)
channel.content_owner => "FullScreen"
channel.time_linked => 2013-03-14 01:14:54 UTC

# channel not in FullScreen
channel = Yt::Channel.new(id: 'UCcMTZY1rFXO3Rj44D5VMyiw', auth: content_owner)
channel.content_owner => nil
channel.time_linked => nil
```

Thoughts?

### CODE REVIEW

**Comment by claudiofullscreen** [comment]
Oh yes, so many attributes still missing. I swear I'll write some meta-programming to automatically add all the attributes, just wait for it :grimacing: 

Meanwhile I'll check what's failing on this build

**Comment by claudiofullscreen** [comment]
@stavro For now I've just rebased off master and `push --force` your commit.
Unfortunately YouTube API is still having problems today, so I cannot fully test this PR :crying_cat_face: 
Let's hope they can fix their issues today!

File: lib/yt/models/content_owner_detail.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      # @return [String] The ID of the content owner linked to the channel.
+      def content_owner
+        @content_owner ||= @data['contentOwner']
+      end
+
+      # @return [String] The date and time of when the channel was linked to the
+      #   content owner.
+      def time_linked
+        @time_linked ||= Time.parse(@data['timeLinked']) rescue nil
+      end
```
Comment (claudiofullscreen) [inline lib/yt/models/content_owner_detail.rb:44]: Let's call this `linked_at`, shall we? :neckbeard:

File: spec/requests/as_content_owner/channel_spec.rb [hunk 2]
```diff
@@ -263,13 +263,20 @@
...
     context 'not managed by the authenticated Content Owner' do
       let(:id) { 'UCBR8-60-B28hp2BmDPdntcQ' }
 
-      it { expect{channel.earnings}.to raise_error Yt::Errors::Forbidden }
-      it { expect{channel.views}.to raise_error Yt::Errors::Forbidden }
+      specify 'information about its content owner cannot be retrieved' do
+        expect{channel.content_owner}.to raise_error Yt::Errors::Forbidden
+        expect{channel.linked_at}.to raise_error Yt::Errors::Forbidden
+      end
     end
```
Comment (claudiofullscreen) [inline spec/requests/as_content_owner/channel_spec.rb:285]: This is quite straightforward: if you are logged in as a CMS owner, but you try to retrieve the `content_owner` of a channel not owned by you, then you get `Yt::Errors::Forbidden`.

File: spec/requests/as_account/channel_spec.rb [hunk 3]
```diff
@@ -105,6 +105,11 @@
       expect{channel.viewer_percentages}.not_to raise_error
       expect{channel.viewer_percentage}.not_to raise_error
     end
+
+    it 'cannot give information about its content owner' do
+      expect{channel.content_owner}.to raise_error Yt::Errors::Forbidden
+      expect{channel.linked_at}.to raise_error Yt::Errors::Forbidden
+    end
```
Comment (claudiofullscreen) [inline spec/requests/as_account/channel_spec.rb:112]: @stavro How do you feel about this?

In short, if you are logged-in as a normal account (not a CMS owner) and try to retrieve the `content_owner` of a channel, you cannot access it, so YT raises `Yt::Errors::Forbidden`. 

In my opinion, this is more meaningful to YouTube behavior, which lets you _ask_ for the `contentOwnerDetails` part but then does not give it back to you if you are not a content owner. For instance:

```
curl -X GET -H "authorization: Bearer ..." "https://www.googleapis.com/youtube/v3/channels?id=UC...&part=contentOwnerDetails"
```

returns

```
{
 "kind": "youtube#channelListResponse",
 "etag": "\"gM...\"",
 "pageInfo": {
  "totalResults": 1,
  "resultsPerPage": 1
 },
 "items": [
  {
   "kind": "youtube#channel",
   "etag": "\"gM...\"",
   "id": "UC..."
  }
 ]
}
```

Basically, you are only asking YouTube for the `contentOwnerDetails` part of the channels, and YouTube does not give you any part back. The way in which I interpret this is: that part is _forbidden_ to you, therefore instead of just letting `channel.content_owner` and `channel.linked_at` return `nil` in the case above (which is confusing), I let Yt specifically raise `Forbidden`.

File: lib/yt/models/request.rb [hunk 4]
```diff
@@ -28,6 +28,7 @@ def initialize(options = {})
       end
 
       def run
+        p as_curl
```
Comment (stavro) [inline lib/yt/models/request.rb]: I've been adding this during my testing also... haha. I don't think you meant to commit it though

File: spec/requests/as_account/channel_spec.rb (hunk 3 above)
Comment (stavro) [inline spec/requests/as_account/channel_spec.rb:112]: Raising a `Forbidden` exception sounds fair to me

File: lib/yt/models/request.rb (hunk 4 above)
Comment (claudiofullscreen) [inline lib/yt/models/request.rb]: Good catch! And yes, I have to add a logger option

**Comment by claudiofullscreen** [comment]
@stavro This PR will be pending until tests pass again, that is, until YouTube restores proper access the content-owner endpoints

**Comment by claudiofullscreen** [comment]
@stavro this is now in ~> 0.9.8
